src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const res = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
```

Comment:
you can use switch for better readability

Reply:
Agree